platform/mbed_sleep.h

```diff
@@ -64,6 +64,34 @@ extern "C" {
  * }
  * @endcode
  */
+#ifdef MBED_SLEEP_STATS_ENABLED
+
+#define sleep_manager_lock_deep_sleep()           \
+    do                                            \
+    {                                             \
+        sleep_manager_lock_deep_sleep_internal(); \
+        sleep_tracker_lock(__FILE__, __LINE__);   \
+    } while (0);
+
+#define sleep_manager_unlock_deep_sleep()           \
+    do                                              \
+    {                                               \
+        sleep_manager_unlock_deep_sleep_internal(); \
+        sleep_tracker_unlock(__FILE__, __LINE__);   \
+    } while (0);
+
+void sleep_tracker_lock(const char *const filename, int line);
```

Comment:
Why do we need it here?

Reply:
> It's sort of the other way around, if you use serial then the deep sleep is disabled.

I am thinking of case, when no device/peripheral has locked deep sleep, so device can actually enter deep sleep. Last unlock, sleep_manager_unlock_deep_sleep_internal() - Counter goes down to zero. 
sleep_tracker_unlock(__FILE__, __LINE__);   - This will enable UART. So it will lock deep sleep again, and unlock again. Will this be recursive? Correct me if I am missing anything or interpreting it wrongly.

Reply:
This sounds like it could be an issue, but I've never seen it happen. I've not seen UART lock the deep sleep while running the tracker, perhaps it locks it through some other means.